### PR TITLE
[API] Use `ml-pipeline` service instead of `ml-pipeline-ui`

### LINKED ISSUE #91: Single user pipeline run is having error
**Describe the bug**
When running single user MLX, the pipeline run is returning the below errors:
```
"Error while trying to run pipeline bdddf286-4c26-46b8-9af5-d08aafc2ce1d: 'MaxRetryError' object has no attribute 'body'"
```

However, for component, model, dataset, notebook runs, the pipeline run works fine.

**To Reproduce**
Install kustomize 3.2.1
```shell
git clone https://github.com/machine-learning-exchange/manifests -b mlx-single-user
cd manifests
# run the below command two times if the CRDs take too long to provision.
kustomize build example | kubectl apply -f -
```
Go to <public_ip>:30380/mlx, publish the default pipeline in the view all pipelines and run it. You will see a blank page because the API is returning the above error.

**Expected behavior**

**Screenshots**

**Environment**

 - MLX Version [e.g. 22]

**Additional context**

### PR DESCRIPTION
**Pull request by ckadner**
Let the `kfp` `Client` determine the connection based on the in-cluster configuration, or, based on the `KUBECONFIG`
environment variable when running the MLX API from outside a cluster.

Resolves #91

Signed-off-by: Christian Kadner <ckadner@us.ibm.com>

### CODE REVIEW

**Comment by ckadner** [comment]
@Tomcli @yhwang -- there may be more API code changes required to fully support the multi-user deployments. This PR just makes sure the connection to KFP is being created in a consistent manner. Prior to this PR I had used 3 slightly different approaches and ML Pipeline (UI) service URLs to connect to the KFP API.

File: api/examples/pipelines_api.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def list_pipelines(filter_dict: dict = {}, sort_by: str = None) -> [ApiPipeline]
 def main():
 
     # delete existing pipelines, actually, only pipeline_extensions table and pipelines_extended view table
-    delete_pipeline(pipeline_id="*")
+    # delete_pipeline(pipeline_id="*")
```
Comment (Tomcli) [inline api/examples/pipelines_api.py:426]: so we no longer need to delete pipelines? Is this function used to clean up all the existing pipelines when we bootstrap?

Comment (ckadner) [reply api/examples/pipelines_api.py:426]: This is just the `example` on how to use the pipeline API endpoints. I use those to run my quasi end-to-end testing. I changed this example to keep whatever pipelines where installed and only later remove the `test_*` pipelines that were uploaded by the `example` script `api/examples/pipelines_api.py`

Comment (ckadner) [reply api/examples/pipelines_api.py:426]: no relations to bootstrapping. I still need to update/create the bootstrapping script [#23](https://github.com/machine-learning-exchange/mlx/issues/23#issuecomment-833842623)

Comment (ckadner) [reply api/examples/pipelines_api.py:426]: @Tomcli -- This is just the `example` on how to use the pipeline API endpoints. I use those to run my _quasi end-to-end testing_. I changed this example to keep whichever pipelines were installed already, and only later on I remove the `test_*` pipelines that were uploaded by the `example` script `api/examples/pipelines_api.py`.

This `api/examples/pipelines_api.py` example script is not used by (or has any relation to) the MLX [bootstrapper](https://github.com/machine-learning-exchange/mlx/tree/main/bootstrapper). I still need to update/create the bootstrapping script [#23](https://github.com/machine-learning-exchange/mlx/issues/23#issuecomment-833842623)

**Comment by Tomcli** [comment]
thanks @ckadner , this works on single user mode

/lgtm

**Comment by ckadner** [comment]
Thanks Tommy!

**Pull request merged**